src/api/auth/hooks.tsx

```diff
@@ -0,0 +1,25 @@
+import { useMutation } from "react-query";
+import queryClient from "src/configs/queryClient";
+import { addToLocalStorage } from "src/helpers/localStorage";
+import { useAuthStore } from "src/store/useAuthStore";
+import { getAuthToken } from "./api";
+import type { IAuthResponse } from "./types";
+
+export const useAuthTokenMutation = () => {
+  const { setIsLogged } = useAuthStore();
+
+  return useMutation<IAuthResponse>({
+    mutationKey: ["auth"],
+    mutationFn: async () => {
+      return getAuthToken().then((data) => {
+        return data;
+      });
```

Comment:
same. pay attention to error handling when you work with promise.

Reply:
yeah, I know(( I haven't done error handling due to lack of time. I will try to find time to fix it. thx